### PR TITLE
Bot refactoring

### PR DESCRIPTION
**Pull request by adambbolduc**
I refactored the bot so that he can be used outside main.go
Only effective change is local is no longer a string but a boolean value.  it is true if environment variable local is "true" or false otherwise.

### CODE REVIEW

File: scenariolib/uabot.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+type uabot struct {
+	local          bool
+	conf           *Config
+	scenarioURL    string
+	searchToken    string
+	analyticsToken string
+	random         *rand.Rand
+}
+
+func NewUabot(local bool, conf *Config, scenarioUrl string, searchToken string, analyticsToken string, random *rand.Rand) *uabot {
```
Comment (erocheleau) [inline scenariolib/uabot.go]: `func NewUabot(local bool, scenarioUrl string, searchToken string, analyticsToken string, random *rand.Rand) *uabot {}`

Instead. And I would move the read file logic as a non-exposed function inside the bot too.

File: main.go [hunk 2]
```diff
@@ -49,17 +45,16 @@ func main() {
...
-	timeNow := time.Now()
 
 	var conf *scenariolib.Config
 	var err error
 	// Init from path instead of URL, for testing purposes
-	if local == "true" {
+	if local {
 		conf, err = scenariolib.NewConfigFromPath(scenarioURL)
 	} else {
 		conf, err = scenariolib.NewConfigFromURL(scenarioURL)
```
Comment (erocheleau) [inline main.go]: These should be moved inside the bot files also.

**Comment by erocheleau** [comment]
I'm not against the idea and I think I would also transfer the logic of reading the config file to the bot also so that your other "main" doesn't have to know the existence of the config class.

So I would init a bot with the path to the config only instead of the actual object.

*See line comments*

File: scenariolib/uabot.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			Info.Println("Updating Time Visits to", *timeVisits)
+		}
+	}()
+}
+
+func (bot*uabot) continuallyRefreshScenariosEvery(timeDuration time.Duration, conf *Config) {
+	ticker := time.NewTicker(timeDuration)
+	go func() {
+		for _ = range ticker.C {
+			conf2 := refreshScenarios(bot.scenarioURL, !bot.local)
```
Comment (erocheleau) [inline scenariolib/uabot.go]: `!bot.local` is pretty confusing here, I would keep the same boolean logic isLocal instead of isUrl.

**Comment by erocheleau** [comment]
Did you test it? maybe change the timers to 1 minute to see what happens when the time between actions change?

**Comment by erocheleau** [comment]
Sorry I don't have any unit tests on this, was not planned like that at first :(

**Comment by adambbolduc** [comment]
Haha, yes it is test with a little timestep to make sure it works !

**Comment by erocheleau** [comment]
Merged, thanks a lot for your help!

**Pull request merged**